src/Common/src/System/Security/Cryptography/ECCng.HashAlgorithm.cs

```diff
@@ -46,5 +48,41 @@ internal static Interop.BCrypt.ECC_CURVE_ALG_ID_ENUM GetHashAlgorithmId(HashAlgo
 
             return new HashAlgorithmName(oid.Name);
         }
+
+        /// <summary>
+        /// Is the curve named, or once of the special nist curves
+        /// </summary>
+        internal static bool IsECNamedCurve(string algorithm)
+        {
+            return (algorithm == BCryptNative.AlgorithmName.ECDH ||
+                    algorithm == BCryptNative.AlgorithmName.ECDsa);
+        }
+
+        /// <summary>
+        /// Maps algorithm to curve name accounting for the special nist curves
+        /// </summary>
+        internal static string SpecialNistAlgorithmToCurveName(string algorithm)
+        {
+            if (algorithm == BCryptNative.AlgorithmName.ECDHP256 ||
+                algorithm == BCryptNative.AlgorithmName.ECDsaP256)
+            {
+                return "nistP256";
+            }
+
+            if (algorithm == BCryptNative.AlgorithmName.ECDHP384 ||
+                algorithm == BCryptNative.AlgorithmName.ECDsaP384)
+            {
+                return "nistP384";
+            }
+
+            if (algorithm == BCryptNative.AlgorithmName.ECDHP521 ||
+                algorithm == BCryptNative.AlgorithmName.ECDsaP521)
+            {
+                return "nistP521";
+            }
```

Comment:
Did you consider using a switch here?

Reply:
This was just a move, but I went ahead and compacted it with switch.